sklearn/utils/testing.py

```diff
@@ -970,3 +975,61 @@ def check_docstring_parameters(func, doc=None, ignore=None, class_name=None):
             if n1 != n2:
                 incorrect += [func_name + ' ' + n1 + ' != ' + n2]
     return incorrect
+
+
+def assert_run_python_script(source_code, timeout=60):
+    """Utility to check assertions in an independent Python subprocess.
+
+    The script provided in the source code should return 0 and not print
+    anything on stderr or stdout.
+
+    This is a port from cloudpickle https://github.com/cloudpipe/cloudpickle
+
+    Parameters
+    ----------
+    source_code : str
+        The Python source code to execute.
+    timeout : int
+        Time in seconds before timeout.
+    """
+    def _make_cwd_env():
+        """Helper to prepare environment for the child processes"""
+        sklearn_repo_folder = op.normpath(
+            op.join(op.dirname(__file__), '../..'))
+        env = os.environ.copy()
+        pythonpath = "{src}{sep}tests{pathsep}{src}".format(
+            src=sklearn_repo_folder, sep=os.sep, pathsep=os.pathsep)
+        env['PYTHONPATH'] = pythonpath
```

Comment:
Actually this pythonpath hack is necessary for cloudpickle because the tests are in `source_repo_folder/tests/`, that is outside of the main package folder.

For scikit-learn, the tests are all under the package folder `source_repo_folder/sklearn` so we probably don't need to pass a custom `PYTHONPATH` environment in our case.

Reply:
We still need to propagate a copy of all other environment variables to the `check_output` command though, but we don't need to forge a custom `PYTHONPATH` variable.